encode.go

```diff
@@ -244,43 +283,49 @@ func (e *encoder) structv(tag string, in reflect.Value) {
 					if _, found := sinfo.FieldsMap[k.String()]; found {
 						panic(fmt.Sprintf("cannot have key %q in inlined map: conflicts with struct field", k.String()))
 					}
-					e.marshal("", k)
+					e.marshal("", k, nil, nil, nil)
 					e.flow = false
-					e.marshal("", m.MapIndex(k))
+					e.marshal("", m.MapIndex(k), nil, nil, nil)
 				}
 			}
 		}
 	})
 }
 
-func (e *encoder) mappingv(tag string, f func()) {
+func (e *encoder) mappingv(tag string, head, line, foot []byte, f func()) {
 	implicit := tag == ""
 	style := yaml_BLOCK_MAPPING_STYLE
 	if e.flow {
 		e.flow = false
 		style = yaml_FLOW_MAPPING_STYLE
 	}
 	yaml_mapping_start_event_initialize(&e.event, nil, []byte(tag), implicit, style)
+	e.event.head_comment = head
```

Comment:
Should we add a comment here that explains that the placement of assignments for `e.event` and the emits have to be done in this particular order?

Reply:
Added Comment!